### DISCUSSION TITLE
反广告兼容 AdGuardHome 的规则吗？

### DISCUSSION THREAD

**Discussion by Skylarzz** [IrineSistiana/mosdns#718] [Q&A]
如果兼容可以考虑代替掉 AdGuardHome

**Comment by sieveLau** [comment]
没有插件支持adblock语法，匹配器语法比较类似v2ray的

**Reply by Skylarzz** [reply]
AdGuardHome 也是基于 DNS，看一下下面这些列表兼容吗？还是只能支持 domain:  full:  keyword: regexp: 开头的域名匹配？

1. AdGuardHome  自带 AdGuard DNS filter `https://adguardteam.github.io/HostlistsRegistry/assets/filter_1.txt`
1. AdGuardHome  自带 AdAway Default Blocklis `https://adguardteam.github.io/HostlistsRegistry/assets/filter_2.txt`
1. Easylist `https://easylist-downloads.adblockplus.org/easylist.txt`
2. EasylistChina `https://easylist-downloads.adblockplus.org/easylistchina.txt`
3. EasyPrivacy `https://easylist-downloads.adblockplus.org/easyprivacy.txt`
4. CJX'sAnnoyance `https://raw.githubusercontent.com/cjx82630/cjxlist/master/cjx-annoyance.txt`
5. 乘风 `https://raw.githubusercontent.com/xinggsf/Adblock-Plus-Rule/master/rule.txt`
6. 乘风视频 `https://raw.githubusercontent.com/xinggsf/Adblock-Plus-Rule/master/mv.txt`
7. Adblock Warning Removal List  `https://easylist-downloads.adblockplus.org/antiadblockfilters.txt`
8. ABP filters `https://easylist-downloads.adblockplus.org/abp-filters-anti-cv.txt`
9. anti-adblock-killer-filters `https://raw.githubusercontent.com/reek/anti-adblock-killer/master/anti-adblock-killer-filters.txt`
10. WindowsSpyBlocker - Hosts spy rules  `https://adguardteam.github.io/HostlistsRegistry/assets/filter_23.txt`
11. I don't care about cookies `https://www.i-dont-care-about-cookies.eu/abp/`

**Reply by sieveLau** [reply]
我之前写过一个翻译器，但是结果后来发现adblock语法的列表很多都停止维护或者品质不好，所以我也放弃了使用adblock语法的列表了。如果你确实想要用这些列表，我把我的repo发给你一下，是c++写的转换器。

**Reply by Skylarzz** [reply]
AdGuardHome 毕竟是反广告专业户，通过它应该能找到质量上乘的列表

**Comment by sieveLau** [comment]
这个[repo](https://github.com/sieveLau/rule2mosdns)的235c355f623d64ca5041148f284053f08004c50c的commit

**Reply by Skylarzz** [reply]
还是希望能完整支持这些规则，实在不行只能和 AdGuardHome 一起使用了
